pkg/syncer/metadatasyncer.go

```diff
@@ -885,7 +885,16 @@ func csiPVUpdated(ctx context.Context, newPv *v1.PersistentVolume, oldPv *v1.Per
 	} else {
 		volumeHandle = newPv.Spec.CSI.VolumeHandle
 	}
-	if oldPv.Status.Phase == v1.VolumePending && newPv.Status.Phase == v1.VolumeAvailable && newPv.Spec.StorageClassName == "" {
+
+	// TODO: Revisit the logic for static PV update once we have a specific return code from CNS
```

Comment:
Can you file an internal bug to track this?

Reply:
I filed an issue in GitHub - https://github.com/kubernetes-sigs/vsphere-csi-driver/issues/579. Can you add a link to this issue here?